File: frontend/src/app/shared/GroupDto.ts
import { UserDto } from './UserDto'
import { IDisplayName } from './IDisplayName'

export class GroupDto implements IDisplayName {
  id: number
  name: string
  description: string

  parentGroupId: number
  parentGroup: GroupDto
  childrenGroups: GroupDto[] = []
  users: UserDto[] = []

  public displayName(): string {
    return this.name
  }
}

File: backend/src2/SharedModels/DTOs/GroupDto.cs
using System.Collections.Generic;

namespace SharedModels.DTOs
{
    public class GroupDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public int? ParentGroupId { get; set; }

        public GroupDto ParentGroup { get; set; }
        public virtual List<UserDto> Users { get; set; }
    }
}